### ISSUE TITLE
Azure.Deployment.AdminUsername does not match the pattern

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#1758]
**Description of the issue**

Complex functions such as `if` or `null` may generate false positives on `Azure.Deployment.AdminUsername`.

**Expected behaviour**

Using functions that do not specifically expose or handle secret values such as `if` and `null` are required for some conditional logic. These are common in Bicep code and should work.

**Error output**

```text
The field value '[if(and(empty(parameters('sqlLogin')),parameters('useAADOnlyAuthentication')),null(),parameters('sqlLogin'))]' does not match the pattern '\[p[^\]]+\]'.
```

```text
The field value '[if(and(empty(parameters('sqlLoginPassword')),parameters('useAADOnlyAuthentication')),null(),parameters('sqlLoginPassword'))]' does not match the pattern '\[p[^\]]+\]'.
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.20.0-B0223**
